gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/DatastoreFactory.java

```diff
@@ -16,18 +16,29 @@
 
 package com.google.gcloud.datastore;
 
+import com.google.gcloud.ServiceFactory;
+import java.io.ObjectStreamException;
 
 /**
  * A base class for Datastore factories.
  */
-public abstract class DatastoreFactory {
+public abstract class DatastoreFactory implements ServiceFactory<Datastore, DatastoreOptions> {
+
+  private static final long serialVersionUID = 5037190305022535983L;
 
   private static final DatastoreFactory INSTANCE = new DatastoreFactory() {
```

Comment:
Yes, though there are more examples.

e.g. writing a resumable blob reader (if we want to support systems like AE mapreduce which
don't read the whole file, or consume the complete result, using the same process).